org.eclipse.cpacep/src/org/eclipse/cpacep/test/CPACEPConnectorTest.java

```diff
@@ -0,0 +1,53 @@
+package org.eclipse.cpacep.test;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.File;
+import java.io.IOException;
+import org.eclipse.cpacep.util.CPACEPConnector;
+import org.eclipse.cpacep.util.FileHandler;
+import org.junit.Before;
+import org.junit.Test;
+
+public class CPACEPConnectorTest {
+	//String homePath = "/home/supun/Documents/cpachecker";
+	//String homePath = "/home/supun/Documents/CPAchecker-1.7-unix";
+	String homePath = "/home/travis/build/SupunArunoda/CPAChecker/cpachecker";
+	StringBuilder sb;
+	String lcSpecification;
+	String lcConfiguration;
+	String result;
+
+	String outputPathOriginal = "/tmp/original";
+	File outputDirectory;
+
+	@Before
+	public void setup() {
+		lcSpecification = "default";
+		lcConfiguration = "default";
+		try {
+			outputDirectory = FileHandler.createTempDirectory();
+		} catch (IOException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+		sb = new StringBuilder();
+		sb.append(homePath + File.separator + "scripts" + File.separator + "cpa.sh");
+		sb.append(" -spec " + homePath + File.separator + "config" + File.separator + "specification" + File.separator + lcSpecification + ".spc");
+		sb.append(" -config " + homePath + File.separator + "config" + File.separator + lcConfiguration + ".properties");
+		sb.append(" " + homePath + File.separator + "doc" + File.separator + "examples" + File.separator + "example.c");
+		sb.append(" -outputpath " + outputDirectory);
+		sb.append(" -stats");
+		CPACEPConnector.executeCommand(sb.toString());
+
+	}
+
+	@Test
+	public void testStatistics() {
+		setup();
+		String actualData = FileHandler.readFile(outputDirectory.getAbsolutePath() + File.separator + "Statistics.txt");
+		String expectData = FileHandler.readFile(outputPathOriginal + File.separator + "Statistics.txt");
+		assertEquals(expectData, actualData);
```

Comment:
I already said so by mail, but to have everything in one place:
Comparing the statistics with a different instance of CPAchecker doesn't make sense.
Some values may be different (as you already noticed yourself), and, more importantly, it is not important that any value is equal and we are dependent on some execution of CPAchecker that is not even part of this test!
Tests should always be self-contained.

Instead, I would suggest to check the output of executing CPAchecker from CPACEP for a line containing "Verification result:" .

Reply:
Hi Thomas,
"Verification result:" statement is at before the last statement. Is this consistent for all statistics generated by CPAChecker?

Reply:
Hi Supun, just ignore the position, it's not important to check whether CPAchecker works.